mmseg/models/backbones/hrformer.py

```diff
@@ -0,0 +1,1071 @@
+
+    Args:
+        embed_dims (int): Number of input channels.
+        num_heads (int): Number of attention heads.
+        window_size (tuple[int]): The height and width of the window.
+        bias (bool, optional):  If True, add a learnable bias to q, k, v.
+            Default: True.
+        dropout (float, optional): Dropout ratio of attention weight.
+            Default: 0.0
+        kdim (int, optional): The number of channels of q/k.
+            Default: None
+        kdim (int, optional): The number of channels of v.
+            Default: None
+    """
+
+    def __init__(
+            self,
+            embed_dim,
+            num_heads,
+            window_size=(7, 7),
+            bias=True,
+            dropout=0.0,
+            kdim=None,
+            vdim=None,
+    ):
+        super(MultiheadAttention, self).__init__()
+        self.embed_dim = embed_dim
+        self.kdim = kdim if kdim is not None else embed_dim
+        self.vdim = vdim if vdim is not None else embed_dim
+
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.head_dim = embed_dim // num_heads
+        assert (self.head_dim * num_heads == self.embed_dim
+                ), 'embed_dim must be divisible by num_heads'
+
+        self.k_proj = nn.Linear(self.kdim, embed_dim, bias=bias)
+        self.v_proj = nn.Linear(self.vdim, embed_dim, bias=bias)
+        self.q_proj = nn.Linear(embed_dim, embed_dim, bias=bias)
+        self.out_proj = nn.Linear(embed_dim, embed_dim)
+
+        self.window_size = [window_size] * 2
+        # define a parameter table of relative position bias
+        self.relative_position_bias_table = nn.Parameter(
+            torch.zeros(
+                (2 * self.window_size[0] - 1) * (2 * self.window_size[1] - 1),
+                self.num_heads,
+            ))  # 2*Wh-1 * 2*Ww-1, nH
+        # pairwise relative position for each token inside the window
+        coords_h = torch.arange(self.window_size[0])
+        coords_w = torch.arange(self.window_size[1])
+        coords = torch.stack(torch.meshgrid([coords_h, coords_w]))  # 2, Wh, Ww
+        coords_flatten = torch.flatten(coords, 1)  # 2, Wh*Ww
+        relative_coords = (coords_flatten[:, :, None] -
+                           coords_flatten[:, None, :])  # 2, Wh*Ww, Wh*Ww
+        relative_coords = relative_coords.permute(
+            1, 2, 0).contiguous()  # Wh*Ww, Wh*Ww, 2
+        relative_coords[:, :,
+                        0] += self.window_size[0] - 1  # shift to start from 0
+        relative_coords[:, :, 1] += self.window_size[1] - 1
+        relative_coords[:, :, 0] *= 2 * self.window_size[1] - 1
+        relative_position_index = relative_coords.sum(-1)  # Wh*Ww, Wh*Ww
+        self.register_buffer('relative_position_index',
+                             relative_position_index)
+        trunc_normal_init(self.relative_position_bias_table, std=0.02)
+
+    def forward(
+        self,
+        query,
+        key,
+        value,
```

Comment:
```suggestion
        value
```

Should not add `,` for the last item.